src/core/net/dhcp6_client.cpp

```diff
@@ -178,6 +178,7 @@ void Client::Start(void)
 
 void Client::Stop(void)
 {
+    mTrickleTimer.Stop();
```

Comment:
It's possible that the DHCP6 client is stopped while it's still soliciting. That is the DHCP6 prefix is removed right after the client solicited.
Currently, the issue is revealed because the BR (which is DHCP6 server & client & owner of the DHCP6 prefix) learns the Network Data containing the DHCP6 prefix from another Thread device after reset, and removes the DHCP6 prefix immediately because of Network Data inconsistency. 
Such issue can also occur by a small chance when a DHCP6 client starts soliciting and DHCP6 server removes the prefix at the same time.

Stopping the trickle timer when stopping DHCP6 client resolves this issue.